flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryRestriction.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.table.restrictions;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.expressions.CallExpression;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.expressions.FieldReferenceExpression;
+import org.apache.flink.table.expressions.ResolvedExpression;
+import org.apache.flink.table.expressions.ValueLiteralExpression;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.FunctionDefinition;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableMap;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to transform a flink {@link ResolvedExpression} in a string restriction that can be
+ * send to BigQuery as a row restriction. Heavily based in the Iceberg expression translation
+ * implementation.
+ */
+@Internal
+public class BigQueryRestriction {
+
+    private BigQueryRestriction() {}
+
+    private static final Pattern STARTS_WITH_PATTERN = Pattern.compile("([^%]+)%");
+
+    /** Represents the possible BQ expressions supported for the correspondent flink ones. */
+    public enum Operation {
```

Comment:
This doesn't seem necessary. The ones which aren't supported will be handled in the switch default case.

Reply:
We need the enum to have a `switch` based control structure (the `FunctionDefinitions` are types behind an interface so we can not match them to `case` branches in Java8. 

I will keep the enum for read clarity sake, but I will add a default option for those unsupported operations throwing an exception to the caller.